Release/src/http/client/http_client_asio.cpp

```diff
@@ -1590,7 +1592,7 @@ class asio_context : public request_context, public std::enable_shared_from_this
                 const auto actualReadSize = op.get();
                 if(actualReadSize == 0)
                 {
-                    this_request->report_exception(http_exception("Unexpected end of request body stream encountered before Content-Length satisfied."));
```

Comment:
Here is the error we were hitting during client sharing.

This was happening after we recreated a context with an empty request read buffer.